src/transformers/trainer.py

```diff
@@ -1983,6 +1983,9 @@ def _inner_training_loop(
                     # if loss is nan or inf simply add the average of previous logged losses
                     tr_loss += tr_loss / (1 + self.state.global_step - self._globalstep_last_logged)
                 else:
+                    assert (
```

Comment:
We don't have any `assert` in our codebase, we test a condition and we raise a specific error if that condition is not met

Reply:
Hi @younesbelkada,  Thank you very much. I will keep that in mind in the future PR.